src/vcpkg/commands.applocal.cpp

```diff
@@ -0,0 +1,539 @@
+            {
+                const auto as_utf16 = Strings::to_utf16(source);
+                Checks::check_exit(VCPKG_LINE_INFO,
+                                   m_tlog_file.write(as_utf16.data(), sizeof(wchar_t), as_utf16.size()) ==
+                                       as_utf16.size());
+                static constexpr wchar_t native_newline = L'\n';
+                Checks::check_exit(VCPKG_LINE_INFO, m_tlog_file.write(&native_newline, sizeof(wchar_t), 1) == 1);
+            }
+
+            if (m_copied_files_log)
+            {
+                const auto& native = source.native();
+                Checks::check_exit(VCPKG_LINE_INFO,
+                                   m_copied_files_log.write(native.c_str(), 1, native.size()) == native.size());
+                Checks::check_exit(VCPKG_LINE_INFO, m_copied_files_log.put('\n') == '\n');
+            }
+
+            ReleaseMutex(the_mutant);
+            CloseHandle(the_mutant);
+
+            return did_deploy;
+        }
+
+        Filesystem& m_fs;
+        Path m_deployment_dir;
+        Path m_installed_bin_dir;
+        Path m_installed_bin_parent;
+        WriteFilePointer m_tlog_file;
+        WriteFilePointer m_copied_files_log;
+        std::unordered_set<std::string> m_searched;
+        bool openni2_installed;
+        bool azurekinectsdk_installed;
+        bool magnum_installed;
+        bool qt_installed;
+    };
+}
+
+namespace vcpkg::Commands
+{
+    void AppLocalCommand::perform_and_exit(const VcpkgCmdArguments& args, Filesystem&) const
+    {
+        static constexpr StringLiteral OPTION_TARGET_BINARY = "target-binary";
+        static constexpr StringLiteral OPTION_INSTALLED_DIR = "installed-bin-dir";
+        static constexpr StringLiteral OPTION_TLOG_FILE = "tlog-file";
+        static constexpr StringLiteral OPTION_COPIED_FILES_LOG = "copied-files-log";
+
+        static constexpr CommandSetting SETTINGS[] = {
+            {OPTION_TARGET_BINARY, []() { return msg::format(msgCmdSettingTargetBin); }},
+            {OPTION_INSTALLED_DIR, []() { return msg::format(msgCmdSettingInstalledDir); }},
+            {OPTION_TLOG_FILE, []() { return msg::format(msgCmdSettingTLogFile); }},
+            {OPTION_COPIED_FILES_LOG, []() { return msg::format(msgCmdSettingCopiedFilesLog); }},
+        };
+
+        const CommandStructure COMMAND_STRUCTURE = {
+            "--target-binary=\"Path/to/binary\" --installed-bin-dir=\"Path/to/installed/bin\" --tlog-file="
+            "\"Path/to/tlog.tlog\" --copied-files-log=\"Path/to/copiedFilesLog.log\"",
+            0,
+            0,
+            {{}, SETTINGS, {}},
+            nullptr};
+
+        auto& fs = get_real_filesystem();
+
+        auto parsed = args.parse_arguments(COMMAND_STRUCTURE);
+        const auto target_binary = parsed.settings.find(OPTION_TARGET_BINARY);
+        Checks::check_exit(
+            VCPKG_LINE_INFO, target_binary != parsed.settings.end(), "The --target-binary setting is required.");
```

Comment:
Normally, error messages need to be localized. However, since this is a z- internal command, I think it's ok to not localize this specific function.